docs/design-patterns.mdx

```diff
@@ -139,23 +139,23 @@ pub contract AContract {
     // Define a struct with the same fields as the resource
     pub struct BReportStruct {
         pub var c: UInt64
-        pub var d: string
+        pub var d: String
 
-        init(c: UInt64, d: string) {
+        init(c: UInt64, d: String) {
             self.c = c
             self.d = d
         }
 
     }
 }
 ...
-import A from 0xA
+import AContract from 0xAContract
 
 // Return the struct with a script
-pub fun main(): A.BReport {
+pub fun main(): AContract.BReportStruct {
     let b: AContract.BResource // Borrow the resource
     let report = b.generateReport()
-    return b
+    return report
```

Comment:
Could just be `return b.generateReport()`

Reply:
absolutely